### PR TITLE
types(datepicker): fix datepicker types

### PR DESCRIPTION
**Pull request by seriouslag**
'function' is not a valid type in typescript

### CODE REVIEW

**Review by yeetrium — APPROVED** [review]
🌽

**Comment by aleks-elkin** [comment]
Thank you for contribution!
Unfortunately, this file is autogenerated, so you need to fix it in another place. To be precise, you should change the function `generatePropertyLine` here: https://github.com/Tradeshift/elements/blob/master/scripts/generate-types.js#L13

**Comment by seriouslag** [comment]
> Thank you for contribution! Unfortunately, this file is autogenerated, so you need to fix it in another place. To be precise, you should change the function `generatePropertyLine` here: https://github.com/Tradeshift/elements/blob/master/scripts/generate-types.js#L13

Updated generator script and updated all type files.

**Review by yeetrium — APPROVED** [review]
⚡

File: scripts/generate-types.js [hunk 1]
```diff
@@ -42,24 +51,67 @@ const getHtmlAttributeInterfaces = (typesFileContent, className, properties) =>
 	typesFileContent += `\tclass?: string;${EOL}`;
 	properties.forEach(property => {
 		// HTML attributes can be one of two types: boolean or string
-		const htmlType = property.Type.toLowerCase() === 'boolean' ? property.Type : 'string';
-		typesFileContent += generatePropertyLine(property.Attribute, htmlType, property.Description);
+		// const htmlType = property.Type.toLowerCase() === 'boolean' ? property.Type : 'string';
+		const tsType = jsTypeToTypescriptType(property.Type);
+		typesFileContent += generatePropertyLine(property.Attribute, tsType, property.Description);
```
Comment (aleks-elkin) [inline scripts/generate-types.js:56]: Please, revert these changes. According to HTML spec, the accepted types for html attributes are string or boolean.
Actually, you can test yourself with a date-picker's function `isDisabledDate`, this code will not trigger the function:
```ts
  const func = () => {
    console.log('called!');
  }

<ts-date-picker isDisabledDate={func}></ts-date-picker>
```

I think the correct way is to do it via ref:
```ts
import React, {useEffect, useRef} from "react";
import "@tradeshift/elements.date-picker";
import {TSDatePicker} from "@tradeshift/elements.date-picker";

function DatePicker() {
  const dateRef = useRef<TSDatePicker & HTMLElement>(null);
  const func = () => {
    console.log('called!');
  }
  useEffect(() => {
    if (dateRef && dateRef.current) {
      dateRef.current.isDisabledDate = func;
    }
  }, [dateRef]);
  return <ts-date-picker ref={dateRef}></ts-date-picker>;
}
```

**Review by aleks-elkin — CHANGES_REQUESTED** [review]

File: scripts/generate-types.js (hunk 1 above)
Comment (seriouslag) [reply scripts/generate-types.js]: Good catch!

Updated.
I changed 'boolean' to be of a similar type as React does for HTML attributes

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L54

**Review by aleks-elkin — APPROVED** [review]
LGTM!
